Bump dependencies stream

rooftopcellist opened the pull request:
We are upgrading Postgresql from 12 to 13 and as such need to handle upgrades and data migration for managed postgres deployments.  This is one potential approach.

rooftopcellist committed:
Bump Postgresql, Nginx and Redis versions
* pg12 --> pg13 upgrade path
* Set supported pg version as a variable to remain DRY
* Make deleting the old db data pvc after upgrade configurable
* Use labels to find the postgres pod

dsavineau commented on roles/installer/tasks/upgrade_postgres.yml:
```diff
@@ -0,0 +1,138 @@
+    command: |
+      bash -c """
+      set -e -o pipefail
+      PGPASSWORD={{ awx_postgres_pass }} {{ pgdump }} | PGPASSWORD={{ awx_postgres_pass }} {{ pg_restore }}
```
I'm not sure if the k8s_exec module supports passing env variables but it would be great to define `PGPASSWORD` as an env var rather than on the cli (if possible)

rooftopcellist replied:
@dsavineau unfortunately it does not support it.  If I add the following to that task, it sets it inside the operator container which is executing the role's tasks, not within the postgresql container which is where the env var is needed.  

```
  environment:
    PGPASSWORD: {{ awx_postgres_pass }}
```

dsavineau committed:
backup/restore: fix postgres label selector value

We need to use the deployment_name variable for the postgres instance
name.

Signed-off-by: Dimitri Savineau <dsavinea@redhat.com>

dsavineau committed:
backup/restore: add missing default supported_pg_version variable

Signed-off-by: Dimitri Savineau <dsavinea@redhat.com>

dsavineau committed:
restore: update database_host fact with pg suffix

Signed-off-by: Dimitri Savineau <dsavinea@redhat.com>

shanemcd commented on README.md:
````diff
@@ -1148,6 +1148,21 @@ To upgrade AWX, it is recommended to upgrade the awx-operator to the version tha
+
+```yaml
+  spec:
+    postgres_keep_pvc_after_upgrade: True
````
Shouldn't this example say `False`?

shanemcd commented on README.md:
````diff
@@ -1148,6 +1148,21 @@ To upgrade AWX, it is recommended to upgrade the awx-operator to the version tha
+```
+
+# Specify whether or not to keep the old PVC after PostgreSQL upgrades
+postgres_keep_pvc_after_upgrade: True
````
Can these 2 line get deleted? Seems redundant.

shanemcd commented on roles/restore/tasks/secrets.yml:
```diff
@@ -30,8 +30,8 @@
-        database_host: "{{ deployment_name }}-postgres"
-      no_log: "{{ no_log }}"
+        database_host: "{{ deployment_name }}-postgres-{{ supported_pg_version }}"
+      no_log: true
```
Don't we want to keep `"{{ no_log }}"`?

shanemcd commented:
Shouldn't we replace all occurrences of `no_log: true` with `"no_log: {{ no_log }}"`?

rooftopcellist commented:
@shanemcd Yes, good catch.  I have replaced all of the no_log lines and removed the redundant doc lines.  I also changed the doc example of `postgres_keep_pvc_after_upgrade` to be the opposite of the default value as suggested.
